### DISCUSSION TITLE
next/image should support running behind a proxy

### DISCUSSION THREAD

**Discussion by kungozan** [vercel/next.js#33872] [Ideas]
### Describe the feature you'd like to request

With the recent introduction of HTTP agent config through the very beneficial `keepAlive` (https://nextjs.org/docs/api-reference/next.config.js/disabling-http-keep-alive) being enabled by default Next.js is starting to become opinionated about how fetch agents should work as a default.

With this in mind I think it would also be a good time to **introduce proxy configuration options to allow `next/image` (and other future framework layer HTTP requests) to be ran from behind a proxy.**

The primary use-case is for applications forced to run behind an egress proxy to validate outgoing requests.

Issues/discussions describing the same problem:
- https://github.com/vercel/next.js/issues/20837
- https://github.com/vercel/next.js/discussions/18376
- https://github.com/vercel/next.js/discussions/21874

## Solution 1 (Preferred)

Use the widely accepted HTTP_PROXY and HTTPS_PROXY variables when configuring the fetch agent.

## Solution 2

An extension of the existing `httpAgentOptions` to include a proxy configuration, e.g in `next.config.js`

```js
module.exports = {
  httpAgentOptions: {
    keepAlive: false,
    proxy: {
        protocol: 'https',
        host: 'https://my-proxy-address.com',
        port: 443
  },
}
```

### Describe alternatives you've considered

An alternative solution is [to override the HTTP agent globally like described here](https://github.com/vercel/next.js/issues/20837) but with this solution it will also [overwrite the keep alive that the Next.js framework introduces with great benefit](https://github.com/gajus/global-agent#what-is-the-reason-global-agent-overrides-explicitly-configured-https-agent).
Without access to the framework I don't see how we can merge agent configs on a per-option basis in a good way, hence the global overwriting, resulting in overriding beneficial behaviours.

**Comment by jernkuan** [comment]
Would this work?

https://github.com/vercel/next.js/issues/47039#issuecomment-1650830576
